### PR TITLE
refactor setup.py

### PR DESCRIPTION
**Pull request by cprogrammer1994**
tested on windows with python 3.5 and 3.6

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,424 +1,228 @@
...
+}))
+
+sources.extend(rglob('examples/OpenGLWindow/**/*.cpp', exclude = {
+	'EGLOpenGLWindow.cpp',
+	'MacOpenGLWindow.cpp',
+	'Win32OpenGLWindow.cpp',
+	'Win32Window.cpp',
+	'X11OpenGLWindow.cpp',
+}))
+
```
Comment (cprogrammer1994) [inline setup.py]: produces a list of 329 files (same as the previous ones)

**Comment by cprogrammer1994** [comment]
I use [this method](https://github.com/cprogrammer1994/ModernGL/blob/master/setup.py#L4) to get the platform, but I am not able to test on Mac.

**Comment by erwincoumans** [comment]
Thanks for some of the work. I'd rather use the original 
"from distutils.util import get_platform". Why does it need to be changed?

File: setup.py [hunk 2]
```diff
@@ -1,424 +1,228 @@
-
-from setuptools import find_packages
+from setuptools import setup, Extension, find_packages
```
Comment (cprogrammer1994) [inline setup.py]: [distutils_vs_setuptools](https://www.reddit.com/r/Python/comments/3882rh/distutils_vs_setuptools/)

Comment (cprogrammer1994) [inline setup.py]: [PyPA sample uses setuptools as well](https://github.com/pypa/sampleproject/blob/master/setup.py#L9)

> setuptools provides enhancements to distutils, and is much more commonly used than just using distutils by itself. _[from distutils docs](https://packaging.python.org/key_projects/#distutils)_

**Comment by erwincoumans** [comment]
I am maintaining setup.py and prefer to make the changes myself. Thanks for your contibution though, I'll have a look at the changes and may manually improve the setup.py eventually!

**Pull request closed without merging**